jms/src/test/scala/akka/stream/alpakka/jms/scaladsl/JmsConnectorsSpec.scala

```diff
@@ -873,42 +873,35 @@ class JmsConnectorsSpec extends JmsSpec with MockitoSugar {
       result.futureValue should contain allElementsOf in
     }
 
-    "produce elements in order" in {
-      val factory = mock[ConnectionFactory]
-      val connection = mock[Connection]
-      val session = mock[Session]
-      val producer = mock[MessageProducer]
+    "produce elements in order" in withMockedProducer { ctx =>
+      import ctx._
+      val delays = new AtomicInteger()
       val textMessage = mock[TextMessage]
 
       val delayedSend = new Answer[Unit] {
-        override def answer(invocation: InvocationOnMock): Unit =
+        override def answer(invocation: InvocationOnMock): Unit = {
+          delays.incrementAndGet()
           Thread.sleep(ThreadLocalRandom.current().nextInt(1, 10))
+        }
       }
 
-      when(factory.createConnection()).thenReturn(connection)
-      when(connection.createSession(anyBoolean(), anyInt())).thenReturn(session)
-      when(session.createProducer(any[javax.jms.Destination])).thenReturn(producer)
       when(session.createTextMessage(anyString())).thenReturn(textMessage)
-      when(producer.send(any[Message], anyInt(), anyInt(), anyLong())).thenAnswer(delayedSend)
+      when(producer.send(any[javax.jms.Destination], any[Message], anyInt(), anyInt(), anyLong()))
+        .thenAnswer(delayedSend)
 
       val in = (1 to 50).map(i => JmsTextMessage(i.toString))
       val jmsFlow = JmsProducer.flow[JmsTextMessage](JmsProducerSettings(factory).withQueue("test").withSessionCount(8))
 
       val result = Source(in).via(jmsFlow).toMat(Sink.seq)(Keep.right).run()
 
       result.futureValue shouldEqual in
+      delays.get() shouldBe 50
```

Comment:
`delays.get()` does not have any side-effect, right? Should it rather be spelled `delays.get shouldBe 50`?